### ISSUE TITLE
Computed property not changing

### ISSUE THREAD

**Issue by xxRockOnxx** [vuejs/vue#3953]
I have asked at Vue's gitter room and they said it looks good.

I can verify the prop my component is receiving is changing by looking at vue's dev-tool. The problem is the computed property does not change. Here's my component:

```
export default {
  name: 'PosTable',
  props: ['items'],

  computed: {
    empty: function (newItems) {
      return Object.keys(newItems).length === 0
    }
  },

  methods: {
    ...mapActions([
      'addToPos'
    ])
  }
}
```

`items` props is from the `store` being passed down by the parent.

Even tried moving the computed to the `getters` and used the store value directly. Same thing, always `true`

**Comment by fnlctrl** [comment]
Getter-only computed property doesn't receive arguments.
Please re-read this section: http://vuejs.org/guide/computed.html

Fix:
```js
computed: {
    empty: function () {
      return Object.keys(this.items).length === 0
    }
  },
```

**Comment by xxRockOnxx** [comment]
Ooops sorry, that was the last suggestion from one of the users at Gitter. That's how it was originally.

**Comment by xxRockOnxx** [comment]
Ok, I give up. I am not sure what's really wrong here.

Things that I have tried:
- compute from `computed` basing from the props
- compute from `getters` basing from the value directly in the store
- compute from the element itself

Only thing that works: turn off caching.

WHAT exactly is the problem? Store value changing means I did something reactive, why are the stuffs depending on Vue's reactivity does not work??
